performance/objcinterop/src/main/kotlin-native/org/jetbrains/objCinteropBenchmarks/complexNumbers.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.complexNumbers
+import kotlinx.cinterop.*
+import platform.posix.*
+import kotlin.math.sqrt
+import kotlin.math.PI
+import kotlin.math.cos
+import kotlin.math.sin
+import kotlin.random.Random
+import platform.Foundation.*
+import platform.darwin.*
+
+actual class ComplexNumbersBenchmark actual constructor() {
+    val complexNumbersSequence = generateNumbersSequence()
+
+    fun randomNumber() = Random.nextDouble(0.0, benchmarkSize.toDouble())
+
+    actual fun generateNumbersSequence(): List<Any> {
+        val result = mutableListOf<Complex>()
+        for (i in 1..benchmarkSize) {
+            result.add(Complex(randomNumber(), randomNumber()))
+        }
+        return result
+    }
+
+    actual fun sumComplex() {
+        complexNumbersSequence.map { (it as Complex).add(it) }.reduce { acc, it -> (acc as Complex).add(it) }
```

Comment:
Is cast required here?

Reply:
Yes, because I had to use Any type if I want to use this method also as benchmark.

Reply:
Well, `Sequence<Complex>` is `Sequence<Any>` too, so you can use trivial wrapper as a benchmark.

Reply:
I was suggested to use better solution `actual typealias`.